akka-remote/src/main/scala/akka/remote/artery/Association.scala

```diff
@@ -464,20 +496,120 @@ private[remote] class Association(
+      abortQuarantined()
+
+      log.info("Unused association to [{}] removed after quarantine", remoteAddress)
+    }
+  }
+
+  def isRemovedAfterQuarantined(): Boolean =
+    queues(ControlQueueIndex) == RemovedQueueWrapper
+
+  private def cancelQuarantinedIdleTimer(): Unit = {
+    val current = quarantinedIdleTask.get
+    current.foreach(_.cancel())
+    quarantinedIdleTask.compareAndSet(current, None)
+  }
+
+  private def startQuarantinedIdleTimer(): Unit = {
+    cancelQuarantinedIdleTimer()
+    quarantinedIdleTask.set(Some(transport.system.scheduler.scheduleOnce(advancedSettings.StopQuarantinedAfterIdle) {
+      if (associationState.isQuarantined())
+        abortQuarantined()
+    }(transport.system.dispatcher)))
+  }
+
+  private def abortQuarantined(): Unit = {
+    streamMatValues.get.foreach {
+      case (queueIndex, OutboundStreamMatValues(killSwitch, _, _)) ⇒
+        killSwitch match {
+          case OptionVal.Some(k) ⇒
+            setStopReason(queueIndex, OutboundStreamStopQuarantinedSignal)
+            clearStreamKillSwitch(queueIndex, k)
+            k.abort(OutboundStreamStopQuarantinedSignal)
+          case OptionVal.None ⇒ // already aborted
+        }
+    }
+  }
+
   private def cancelIdleTimer(): Unit = {
-    val current = idle.get
+    val current = idleTask.get
     current.foreach(_.cancel())
-    idle.compareAndSet(current, None)
+    idleTask.compareAndSet(current, None)
   }
 
   private def startIdleTimer(): Unit = {
     cancelIdleTimer()
-    idle.set(Some(transport.system.scheduler.scheduleOnce(advancedSettings.StopQuarantinedAfterIdle) {
-      if (associationState.isQuarantined())
-        streamMatValues.get.valuesIterator.foreach {
-          case OutboundStreamMatValues(killSwitch, _) ⇒ killSwitch.abort(OutboundStreamStopSignal)
+    val StopIdleOutboundAfter = settings.Advanced.StopIdleOutboundAfter
+    val interval = StopIdleOutboundAfter / 2
+    val stopIdleOutboundAfterNanos = StopIdleOutboundAfter.toNanos
+    val initialDelay = (settings.Advanced.ConnectionTimeout).max(StopIdleOutboundAfter) + 1.second
+    val task: Cancellable = transport.system.scheduler.schedule(initialDelay, interval) {
+      if (System.nanoTime() - associationState.lastUsedTimestamp.get >= stopIdleOutboundAfterNanos) {
+        streamMatValues.get.foreach {
+          case (queueIndex, OutboundStreamMatValues(streamKillSwitch, _, stopping)) ⇒
+            if (isStreamActive(queueIndex) && stopping.isEmpty) {
+              if (queueIndex != ControlQueueIndex) {
+                streamKillSwitch match {
+                  case OptionVal.Some(k) ⇒
+                    // for non-control streams we can stop the entire stream
+                    log.info("Stopping idle outbound stream [{}] to [{}]", queueIndex, remoteAddress)
+                    setStopReason(queueIndex, OutboundStreamStopIdleSignal)
+                    clearStreamKillSwitch(queueIndex, k)
+                    k.abort(OutboundStreamStopIdleSignal)
```

Comment:
Does this lead to a flight recorder event somewhere? Should it?

Reply:
added that and a few more